src/main/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingImpl.java

```diff
@@ -0,0 +1,116 @@
+package se.liu.ida.hefquin.engine.data.impl;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Set;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.query.Query;
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QueryExecutionFactory;
+import org.apache.jena.query.QueryFactory;
+import org.apache.jena.query.QuerySolution;
+import org.apache.jena.query.ResultSet;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.Property;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.reasoner.TriplePattern;
+import org.apache.jena.riot.RDFDataMgr;
+
+import se.liu.ida.hefquin.engine.data.VocabularyMapping;
+
+public class VocabularyMappingImpl implements VocabularyMapping{
+	
+	protected final Model vocabularyMapping;
+	
+	public VocabularyMappingImpl() {
+		vocabularyMapping = ModelFactory.createDefaultModel();
+	}
+	
+	
+	// Source: https://jena.apache.org/documentation/io/rdf-input.html
+	public VocabularyMappingImpl(final String rdfFile) {
+		vocabularyMapping = RDFDataMgr.loadModel(rdfFile); //.nt file for N-Triple
+	}
+	
+	public VocabularyMappingImpl(final Set<TriplePattern> tps) {
+		vocabularyMapping = ModelFactory.createDefaultModel();
+		Iterator<TriplePattern> i = tps.iterator();
+		while(i.hasNext()) {
+			TriplePattern tp = i.next();
+			Resource s = vocabularyMapping.createResource(tp.getSubject().toString());
+			Property p = vocabularyMapping.createProperty(tp.getPredicate().toString());
+			RDFNode o = vocabularyMapping.createResource(tp.getObject().toString());
+			Statement statement = vocabularyMapping.createStatement(s, p, o);
+			vocabularyMapping.add(statement);
+		}
+	}
+
+	@Override
+	//TODO: Which data structure to use? Unions and intersections also need to be represented
+	public Set<TriplePattern> translateTriple(TriplePattern tp) {
+		Set<TriplePattern> results = translateSubject(tp);
+		return results;
+	}
+	
+	private Set<TriplePattern> translateSubject(TriplePattern tp){
+		Set<TriplePattern> results = Collections.emptySet();
+		if (tp.getSubject().isVariable()) {
```

Comment:
This does not work. Our `TriplePattern` interface does not have a function called `getSubject()`. You have to do:
```suggestion
		if ( tp.asJenaTriple().getSubject().isVariable() ) {
```

...ah, now I see that you have imported the from `TriplePattern`. The one that you need to use is the one that we have in the HeFQUIN code, in the Java package `se.liu.ida.hefquin.engine.query`.